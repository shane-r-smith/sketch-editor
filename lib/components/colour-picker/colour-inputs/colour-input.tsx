import type { BaseColourPickerProps } from "../colour-picker.types";
import { HexColourInput } from "./hex-colour-input";
import { RgbColourInput } from "./rgb-colour-input";

export function ColourInput({ type, ...rest }: BaseColourPickerProps) {
  if (type === "RGB") {
    return <RgbColourInput {...rest} type={type} />;
  }

  return <HexColourInput {...rest} type={type} />;
}
