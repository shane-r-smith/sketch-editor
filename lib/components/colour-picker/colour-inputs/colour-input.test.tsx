import { render } from "@testing-library/react";
import { vi } from "vitest";
import { ColourInput } from "./colour-input";
import type { BaseColourPickerProps } from "../colour-picker.types";

vi.mock("./hex-colour-input", () => ({
  HexColourInput: (props: BaseColourPickerProps) => (
    <div>{JSON.stringify(props, undefined, 2)}</div>
  ),
}));
vi.mock("./rgb-colour-input", () => ({
  RgbColourInput: (props: BaseColourPickerProps) => (
    <div>{JSON.stringify(props, undefined, 2)}</div>
  ),
}));

describe("<ColourInput />", () => {
  const baseProps: BaseColourPickerProps = {
    colour: "#ff0000",
    onChange: vi.fn(),
    type: "HEX" as const,
  };

  it("renders HexColourInput when type is HEX", () => {
    const { container } = render(<ColourInput {...baseProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders RgbColourInput when type is RGB", () => {
    const props = { ...baseProps, type: "RGB" as const };
    const { container } = render(<ColourInput {...props} />);
    expect(container).toMatchSnapshot();
  });
});
