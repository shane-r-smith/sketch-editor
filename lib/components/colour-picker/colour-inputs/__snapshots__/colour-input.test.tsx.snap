// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ColourInput /> > renders HexColourInput when type is HEX 1`] = `
<div>
  <div>
    {
  "colour": "#ff0000",
  "type": "HEX"
}
  </div>
</div>
`;

exports[`<ColourInput /> > renders RgbColourInput when type is RGB 1`] = `
<div>
  <div>
    {
  "colour": "#ff0000",
  "type": "RGB"
}
  </div>
</div>
`;
